x-pack/plugins/elastic_assistant/server/lib/langchain/execute_custom_llm_chain/index.ts

```diff
@@ -25,13 +25,15 @@ export const callAgentExecutor = async ({
   langChainMessages,
   logger,
   request,
+  elserId,
 }: {
   actions: ActionsPluginStart;
   connectorId: string;
   esClient: ElasticsearchClient;
   langChainMessages: BaseMessage[];
   logger: Logger;
   request: KibanaRequest<unknown, unknown, RequestBody>;
+  elserId: string;
```

Comment:
Since this isn't optional, that means `await plugins.ml.trainedModelsProvider(request, savedObjectsClient).getELSER()` will always return a model id, right? IIRC from https://github.com/elastic/kibana/pull/167000, this returns the default/recommended model, and so might not be downloaded/deployed?

This should all be fine, just want to confirm behavior 🙂

Reply:
Yes, it might not be downloaded/deployed yet

Reply:
Now I have some doubts about the logic here. What if the recommended model for the cluster is .elser_model_2_linux-x86_64` which is _not_ downloaded/deployed yet, but `.elser_model_2`has already been deployed? Should we use the second one instead? 🤔

Reply:
So I could see it going either way... There's utility in maintaining the 'recommended/default' as perhaps it offers the best performance/compatibility/whatever, but if we're consuming this API to determine a 'friendly default' that'll just work for the user, then maybe we do want to return the model that is already deployed and will still work. Otherwise consumers (security assistant here) will need to manage some sort of manual override by the user for the scenario you described, or the user will _have to setup/deploy_ the recommended model.

I think we were planning on offering a 'model selector' UI in the settings for folks wanting to specify the model, and to also surface the upcoming Inference API as an option as well, so if you want to keep it as-is for now, I don't think that'll be an issue in the interim.